### PR TITLE
feat(backend): gym branding, logo upload, gymBoards, and board presence ids

### PR DESCRIPTION
**Pull request by marcodejongh**
## Summary

PR D of the gym-kiosk-dashboard epic — the backend surface the kiosk, embeds, and manage UI build on. **Stacked on #3627** (`feat/gym-kiosk-db-schema`); this PR bases on that branch and will retarget to `main` on merge. The branding columns and `gym_kiosks` table land in #3627; this PR exposes branding on the GraphQL surface and adds the read/write surface.

**What & why:**

- **Gym branding on the GraphQL surface.** `logoUrl`, `brandPrimaryColor`, `brandAccentColor`, `brandBackgroundColor` on `Gym` and `UpdateGymInput`. `logoUrl` is the square kiosk/embed logo, distinct from `imageUrl` (the gym photo). Zod `HexColorSchema` (`#RRGGBB`) and an https-or-`/static/gym-logos/` `GymLogoUrlSchema`. `updateGym` persists all four behind the existing `requireGymEditAccess`; an explicit `null` clears a column, `undefined` leaves it untouched. `enrichGym` returns them.
- **Logo upload — `POST /api/gym-logos`.** Mirrors the avatar handler: Busboy, 2 MB cap, `image/jpeg|png|gif|webp` (no SVG — an inline `<svg>` would run script on the kiosk), zero-byte uploads rejected, Bearer auth, gym loaded by uuid and gated by `userCanEditGym` (403 otherwise, 404 for a missing gym); the whole authz/DB block inside the busboy `finish` listener is try/caught so a transient DB error yields a 500 instead of an unhandled rejection + hung request. S3 key `gym-logos/{gymUuid}.{ext}`, written **before** stale other-extension cleanup (a failed replacement never deletes the existing logo); `./gym-logos` local-dev fallback, 501 in prod without S3. Served at `/static/gym-logos/:file` (resize-on-the-fly, same as avatars) with `X-Content-Type-Options: nosniff`. Returns a versioned `/static/gym-logos/{uuid}.{ext}?v=…` URL; `GymLogoUrlSchema`'s static branch is full-pattern matched (no traversal/other-path strings) and its https branch is `new URL()`-parsed.
- **`gymBoards(gymUuid): [UserBoard!]!` — viewer-scoped public read.** Auth-optional. Gym editors (owner, gym admin/editor, covering community admin/leader) see every non-deleted linked board; everyone else — including anonymous — sees only publicly **listed** boards (`isPublic AND NOT isUnlisted`, matching searchBoards' discovery convention: unlisted = link-only, never enumerated). A missing gym, or a private gym seen by a non-editor, is masked as `NOT_FOUND` (stricter than the legacy `gym`/`gymBySlug` reads — aligning those is a tracked follow-up). Ordered by board name. The leaderboard embed reuses it anonymously, so the query itself is **not** canEdit-gated. Rate-limited (30/min, matching the board-presence anon family it feeds). Kiosks are unaffected by the unlisted exclusion — the kiosk read resolves boards from its layout, not gymBoards.
- **`UserBoard.boardId: Int`.** Numeric board-presence channel id (`userBoards.id`), gated by a single shared helper (`boardPresenceChannelId`) across `enrichBoard`/`enrichBoards`/the anon serial-lookup path: populated only when the board is public or the viewer can edit it, else `null`. Feeds `boardNowPlaying(boardId)` for kiosk/embed surfaces without a resolve mutation; documented in `docs/websocket-implementation.md`.
- Added `GET_GYM_BOARDS` + branding fields on `GYM_FIELDS`; regenerated codegen. **Zero `packages/db` changes.**

`GET_GYM_BOARDS` field list: `uuid, slug, name, boardType, layoutId, sizeId, setIds, angle, isAngleAdjustable, isPublic, isUnlisted, locationName, gymId, gymUuid, boardId, canEdit`.

## Release Notes

- [x] No release note needed (internal / technical change)

none — backend/API groundwork with no user-facing surface yet.

## Deliberately skipped (tracked as future polish)

- Upload rate limiting on `POST /api/gym-logos` (parity with the avatar handler, which also has none).
- Deduplicating the shared Busboy/upload plumbing between `avatars.ts` and `gym-logos.ts` into a common helper.

## Test plan

- `vp check` — my files clean (remaining errors/warnings are pre-existing repo debt in unrelated files).
- `vp run typecheck` — passes across all packages, including web (additive optional fields break no consumer).
- New backend test files (run single-file per the concurrency rule), 39 tests green:
  - `gym-logo-upload.test.ts` — no-auth 401, invalid-token 401, owner/admin/editor/global-community-admin 200, random 403, unknown gym 404, malformed `gymUuid` 400, SVG-mime 400, >2 MB 400, zero-byte 400, missing-`gymUuid` 400, no-file-part 400, DB-failure-in-authz-window → 500 (not a hang), static-route path-traversal filename → 400, S3 write-first ordering (failed replacement never triggers cleanup; success cleans exactly the other extensions), extension-switch re-upload removes the stale local file, plus serve-through-static and persist-via-`updateGym`.
  - `gym-branding-and-boards.test.ts` — branding round-trip + null-clears + hex/non-https rejection; `gymBoards` visibility matrix (editor sees private + unlisted linked boards, anon/non-editor sees listed-public only, unlisted excluded for anon, private gym → `NOT_FOUND`, unknown gym → `NOT_FOUND`); `boardId` nullness matrix (anon+public → int, anon+private → null, owner → int, gym-editor on a private board → null since board-edit is stricter than gym-visibility).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01MaW2zXJkhwXJhyxYzSv9iH

### CODE REVIEW

File: packages/backend/src/handlers/gym-logos.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+      // Determine file extension
+      const ext = MIME_TO_EXT[mimeType] || 'jpg';
+      const logoFileName = `${gymUuid}.${ext}`;
+      let logoUrl: string;
+
+      try {
+        if (useS3) {
+          // Remove any prior logo (other extensions) so a re-upload can't leave
+          // a stale file at a different key.
+          await deleteGymLogosFromS3(gymUuid);
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/backend/src/handlers/gym-logos.ts]: Fixed in b2d8b2992 — the handler now writes the new logo first and cleans up stale other-extension files only after the write succeeds (same-extension re-uploads overwrite their key in place, which is atomic). `deleteGymLogosFromS3`/`deleteExistingLocalLogos` take a `keepExt` so the new file is never touched. Covered by two new tests: a failed S3 upload never triggers cleanup (existing logo + stored `logoUrl` stay intact), and a successful upload cleans exactly the other extensions.

**Comment by marcodejongh** [comment]
Re: [Claude Review — hang on DB failure + rate-limit mismatch](https://github.com/boardsesh/boardsesh/pull/3631#issuecomment-4979730403)

- **Unhandled rejection / request hang in `busboy.on('finish')`**: fixed in 2176245b4 — the gym-load + `userCanEditGym` block is wrapped in try/catch (500 + settle), with a regression test that fails the DB call in exactly that window and asserts a 500 rather than a hang.
- **`gymBoards` rate limit**: adjusted in b2d8b2992 — bumped 20 → 30/min to match the board-presence anon family it feeds (`boardNowPlaying`/`boardPresenceStats`/`boardConnection` are 30/min, not 60 as the review states; `boardRecentClimbs`/`boardHistory` are the 60s). gymBoards is a one-shot enumeration per embed/kiosk load, so 30/min per NAT'd IP comfortably covers a multi-display gym reconnecting after a blip.
- **`ensureGymLogosDir` EEXIST branch**: declined — it deliberately mirrors `avatars.ts` line-for-line (the two handlers are kept diffable; consolidating the shared upload plumbing is the tracked follow-up noted in the PR body).
- **Docs**: added in b2d8b2992 — `docs/websocket-implementation.md` (Board Presence Wall Feed section) now documents `UserBoard.boardId` as the resolve-mutation-free path to `boardNowPlaying` for kiosk/embed surfaces, including the public-or-board-editable gate.

**Comment by marcodejongh** [comment]
Re: [Claude Review — MIME spoofing / SVG bypass, 0-byte file, orphaned uploads](https://github.com/boardsesh/boardsesh/pull/3631#issuecomment-4980025323)

- **SVG-as-jpeg spoofing**: analysed — no path can serve the payload as `image/svg+xml`. The served `Content-Type` comes either from the S3 object metadata (which is the upload's declared type, allowlisted to jpeg/png/gif/webp) or from the extension map (`MIME_TYPES[ext]`, where ext is one of jpg/png/gif/webp) — `image/svg+xml` is unreachable on every branch, including the `?size=` resize path (which outputs JPEG or falls back to the original bytes with the same allowlisted type). An SVG payload served as `image/png` is inert in `` and browsers don't sniff a declared raster type into a scriptable SVG document. Per your alternative suggestion, b2d8b2992 adds `X-Content-Type-Options: nosniff` on every `/static/gym-logos/` response as belt-and-braces for legacy clients/proxies; the byte-sniffing check is declined as redundant given both layers.
- **0-byte file accepted**: fixed in b2d8b2992 — `fileBuffer.length === 0` now returns 400 (`Uploaded file is empty`), with a test.
- **Logo URL not atomically persisted**: acknowledged, deliberate — mirrors the avatar upload contract (upload returns a URL, the client persists via `updateGym`). An orphaned object is overwritten by the next upload at the same `gym-logos/{gymUuid}.{ext}` key, so growth is bounded per gym. Noted as follow-up material alongside the avatars-parity items in the PR body.

**Comment by marcodejongh** [comment]
Re: [Claude Review — silent S3 delete errors, missing test, boardId duplication, docs](https://github.com/boardsesh/boardsesh/pull/3631#issuecomment-4983528465)

- **`deleteGymLogosFromS3` silently eats delete errors**: fixed in b2d8b2992 — S3 `DeleteObject` is idempotent (deleting a missing key succeeds), so anything caught there is a real failure and is now logged with `logger.warn`. `deleteExistingLocalLogos` now scopes its ignore to `ENOENT` and warns on anything else. Both cleanups also moved to AFTER the new logo is written (see the codex thread), so a cleanup failure can only ever leave an unreferenced stale-ext file behind, never break the live logo.
- **Missing "no file uploaded" test**: added in b2d8b2992 (`omitFile` → 400 `No file uploaded`), plus a zero-byte-file 400 case.
- **`boardId` duplication in the anon serial-lookup path**: fixed in b2d8b2992 — extracted `boardPresenceChannelId(board, canEdit)` as the single gate, used by `enrichBoard`, `enrichBoards`, and the anon `boardsBySerialNumbers` builder.
- **Docs**: updated in b2d8b2992 — `docs/websocket-implementation.md` (Board Presence Wall Feed) now documents `UserBoard.boardId` as the resolve-mutation-free path to `boardNowPlaying` and points at the shared gate.
